api/src/controllers/invitationController.ts

```diff
@@ -65,6 +65,10 @@ export const redeemInvitation = async (req: Request, res: Response) => {
   const invitationCode = req.body.invitationCode;
   const userId = req.userId;
 
+  if (!userId) {
+    return;
+  }
+
   const user: User | null = await UserRepository.findUserById(userId);
```

Comment:
we can remove the null here coz we exit the func if !userId so user should never be null.

Reply:
Can't do that. Its the type returned by the repository.

Reply:
I think the repository shouldn't return null when it can't find the user instead throw an error. It should strictly return a User.

Reply:
Thats a good point. We should do the same thing with all the other repositories as well.